INN/umbrella-borderzine
# Search by/for author

## Changes

This pull request makes the following changes:

- Copies `search.php` partial from Largo
- Creates new `content-author-search.php` partial to display author archive links
- Modifies the default WP search query to search for posts written by author if the search term matches an author name in the new `borderzine_search_posts_by_author` function
- Modifies `search.php` template to see if the search term matches an author and if it does, updates the `found_posts` int and calls in the new `content-author-search.php` partial to display a link to the author archive before returned posts

Author archive link:
![Screen Shot 2019-09-26 at 2 33 46 PM](https://user-images.githubusercontent.com/18353636/65715464-221d9f80-e06b-11e9-9931-4273da8db156.png)

## Why



For #25

## Testing/Questions

Features that this PR affects:

- Search page + results


Questions that need to be answered before merging:

- [ ] Currently the functionality to return author archive pages only works if the search term matches an author `display_name`. Should this be expanded more?
- [ ] Is there a better way to do any of this?
- [ ] Any concerns we should have here in terms of sanitization before running the queries?

Steps to test this PR:

1. Search for a post title and verify that works ok
2. Search for an author, example `Molly Hunter`, and verify the author archive link is displayed.
3. Search for an author, example `Molly Hunter`, and verify that posts written by the author are returned.
